controllers/appgroup_controller.go

```diff
@@ -165,6 +176,10 @@ func (r *ApplicationGroupReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 	// Compares the current generation to the generation that was last
 	// seen and updated by the reconciler
 	if appGroup.Generation != appGroup.Status.ObservedGeneration {
+		// Change the app group spec into a progressing state
+		appGroup.Progressing()
+		_ = r.Status().Update(ctx, &appGroup)
```

Comment:
@jonathan-innis This is causing the `appGroup.Status.ObservedGeneration` to get bumped up.
`appGroup.Progressing()` logic below bumps it up.
```go
/ Progressing resets the conditions of the ApplicationGroup to
// metav1.Condition of type meta.ReadyCondition with status 'Unknown' and
// meta.StartingReason reason and message.
func (in *ApplicationGroup) Progressing() {
	in.Status.ObservedGeneration = in.Generation
	in.Status.Conditions = []metav1.Condition{}
	meta.SetResourceCondition(in, meta.ReadyCondition, metav1.ConditionUnknown, meta.ProgressingReason, "workflow is reconciling...")
	meta.SetResourceCondition(in, meta.DeployCondition, metav1.ConditionUnknown, meta.ProgressingReason, "application group is reconciling...")
}
```

This causes the `appGroup.Status.ObservedGeneration` to no longer be 0 on a first time reconcile which in turn enables  the `appGroup.Status.Update` flag which is checked below - leading to the following error during reconcile

```logs
{"level":"error","ts":1618417317.880827,"logger":"controllers.ApplicationGroup","msg":"failed to DELETE argo workflow object","appgroup":"bookinfo","appgroup":"bookinfo","error":"resource name may not be empty","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/nitish_malhotra/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\ngithub.com/Azure/Orkestra/pkg/workflow.(*argo).Submit\n\t/Users/nitish_malhotra/github/azure/orkestra/pkg/workflow/argo.go:180\ngithub.com/Azure/Orkestra/controllers.(*ApplicationGroupReconciler).generateWorkflow\n\t/Users/nitish_malhotra/github/azure/orkestra/controllers/appgroup_reconciler.go:248\ngithub.com/Azure/Orkestra/controllers.(*ApplicationGroupReconciler).reconcile\n\t/Users/nitish_malhotra/github/azure/orkestra/controllers/appgroup_reconciler.go:60\ngithub.com/Azure/Orkestra/controllers.(*ApplicationGroupReconciler).Reconcile\n\t/Users/nitish_malhotra/github/azure/orkestra/controllers/appgroup_controller.go:186\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/nitish_malhotra/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/nitish_malhotra/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\t/Users/nitish_malhotra/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/Users/nitish_malhotra/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/Users/nitish_malhotra/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/Users/nitish_malhotra/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/nitish_malhotra/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/Users/nitish_malhotra/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/Users/nitish_malhotra/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99"}
```